src/pages/index.js

```diff
@@ -0,0 +1,318 @@
+let selectedCardId = null;
+const deleteModal = document.querySelector("#delete-modal");
+
+const handleDeleteCard = (cardElement, cardId) => {
+  if (!cardId) {
+    console.error("Card ID is invalid:", cardId);
+    return;
+  }
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+};
+
+const handleDeleteSubmit = (event) => {
+  event.preventDefault();
+  api
+    .removeCard(selectedCardId)
+    .then(() => {
+      if (selectedCard instanceof HTMLElement) {
+        if (selectedCard instanceof HTMLElement) {
+          selectedCard.remove();
+        } else {
+          console.error(
+            "selectedCard is not a valid DOM element:",
+            selectedCard
+          );
+        }
+      } else {
+        console.error("selectedCard is not a DOM element:", selectedCard);
+      }
+      closeModal(deleteModal);
+    })
+    .catch((err) => console.error("Error deleting card:", err));
+};
+
+// ------------------- CARD CREATION & RENDERING -------------------
+const createCardElement = ({
+  name,
+  link,
+  _id,
+  isLiked = false,
+  likes = [],
+}) => {
+  const cardElement = selectors.cardTemplate.cloneNode(true).firstElementChild;
+
+  const cardImage = cardElement.querySelector(".card__image");
+  const cardTitle = cardElement.querySelector(".card__title");
+  const likeButton = cardElement.querySelector(".card__like-button");
+  const likeCount = cardElement.querySelector(".card__like-count");
+  const deleteButton = cardElement.querySelector(".card__delete-button");
+
+  cardImage.src = link;
+  cardImage.alt = name;
+  cardTitle.textContent = name;
+
+  if (isLiked) {
+    likeCount.textContent = 1;
+    likeButton.classList.add("card__like-button_active");
+  } else {
+    likeCount.textContent = 0;
+  }
+
+  likeButton.addEventListener("click", () => {
+    const isCurrentlyLiked = likeButton.classList.contains(
+      "card__like-button_active"
+    );
+    const apiCall = isCurrentlyLiked ? api.removeLike(_id) : api.addLike(_id);
+
+    apiCall
+      .then((updatedCard) => {
+        const newIsLiked = updatedCard.isLiked;
+        const newLikes =
+          parseInt(likeCount.textContent, 10) + (newIsLiked ? 1 : -1);
+
+        likeButton.classList.toggle("card__like-button_active", newIsLiked);
+        if (likeCount) likeCount.textContent = newLikes;
+      })
+      .catch((err) => console.error("Error updating like:", err));
+  });
+
+  deleteButton.addEventListener("click", () =>
+    handleDeleteCard(cardElement, _id)
+  );
+  cardImage.addEventListener("click", () => openPreviewModal(link, name));
+
+  return cardElement;
+};
+
+const openPreviewModal = (imageSrc, imageAlt) => {
+  selectors.previewImage.src = imageSrc;
+  selectors.previewImage.alt = imageAlt;
+  selectors.previewCaption.textContent = imageAlt;
+  openModal(selectors.previewModal);
+};
+
+const renderCard = (item, method = "prepend") => {
+  if (document.querySelector(`[data-id="${item._id}"]`)) {
```

Comment:
```js

  if (document.querySelector(`[data-id="${item._id}"]`)) {

```


It's not needed because it's a waste of resources on searching an element in the DOM